pkg/controller/constraint/constraint_controller.go

```diff
@@ -211,43 +195,43 @@ func (r *ReconcileConstraint) Reconcile(request reconcile.Request) (reconcile.Re
 	return reconcile.Result{}, nil
 }
 
-func (r *ReconcileConstraint) addConstraintKey(constraintKey string, t Tags) {
-	r.constraintsCacheMux.Lock()
-	defer r.constraintsCacheMux.Unlock()
+func (r *ReconcileConstraint) addConstraintKey(constraintKey string, t util.Tags) {
+	r.constraintsCache.CacheMux.Lock()
+	defer r.constraintsCache.CacheMux.Unlock()
 
-	r.constraintsCache[constraintKey] = Tags{
-		enforcementAction: t.enforcementAction,
-		status:            t.status,
+	r.constraintsCache.Cache[constraintKey] = util.Tags{
+		EnforcementAction: t.EnforcementAction,
+		Status:            t.Status,
 	}
 }
 
 func (r *ReconcileConstraint) deleteConstraintKey(constraintKey string) {
-	r.constraintsCacheMux.Lock()
-	defer r.constraintsCacheMux.Unlock()
+	r.constraintsCache.CacheMux.Lock()
+	defer r.constraintsCache.CacheMux.Unlock()
 
-	delete(r.constraintsCache, constraintKey)
+	delete(r.constraintsCache.Cache, constraintKey)
 }
 
 func (r *ReconcileConstraint) reportTotalConstraints() {
```

Comment:
This should be a method on the constraint cache struct.

The reporter could be passed as an argument.

Reply:
if reporter is passed, it'll be cyclical dependency i think

Reply:
moved back to constraints package